src/components/ComputeTime.tsx

```diff
@@ -0,0 +1,149 @@
+      }
+      let clockField: Field | undefined =
+        data.series[0].fields.find((field: Field) => field.name === clockSettings.queryField) ?? undefined;
+
+      let descriptionField: Field | undefined =
+        data.series[0].fields.find((field: Field) => field.name === options.descriptionSettings.queryField) ??
+        undefined;
+
+      if (!clockField || !clockField.values || clockField.values.length === 0) {
+        return [now, description, clockNoValueText];
+      }
+
+      let descriptionFieldValues = descriptionField?.values ?? new Array(clockField.values.length);
+      let fieldValues: Array<{ time: MomentInput; description: string }> = clockField.values.map((value, index) => {
+        return { time: value, description: descriptionFieldValues[index] };
+      });
+
+      let values: Array<{ time: Moment; description: string }> = fieldValues
+        .filter((v) => v.time !== null && v.time !== undefined && !Number.isNaN(v.time))
+        .map((v) => {
+          return { time: moment(v.time), description: v.description };
+        });
+
+      let sort_values = (
+        v: Array<{ time: Moment; description: string }>
+      ): Array<{ time: Moment; description: string }> => {
+        return v.sort((a, b) => a.time.diff(b.time));
+      };
+
+      let _value: { time: Moment; description: string } | undefined = undefined;
+      switch (clockSettings.queryCalculation) {
+        case QueryCalculation.lastNotNull:
+          _value = values.length > 0 ? values.at(-1) : undefined;
+          break;
+        case QueryCalculation.last:
+          _value =
+            fieldValues.length > 0
+              ? { ...fieldValues[fieldValues.length - 1], time: moment(fieldValues[fieldValues.length - 1]?.time) }
+              : undefined;
+          break;
+        case QueryCalculation.firstNotNull:
+          _value = values.length > 0 ? values[0] : undefined;
+          break;
+        case QueryCalculation.first:
+          _value = fieldValues.length > 0 ? { ...fieldValues[0], time: moment(fieldValues[0].time) } : undefined;
+          break;
+        case QueryCalculation.min:
+          _value = values.length > 0 ? sort_values(values)[0] : undefined;
+          break;
+        case CountdownQueryCalculation.minFuture:
+          values = values.filter((v: { time: Moment; description: string }) => v.time.isAfter(now));
+          _value = values.length > 0 ? sort_values(values)[0] : undefined;
+          break;
+        case QueryCalculation.max:
+          _value = values.length > 0 ? sort_values(values).at(-1) : undefined;
+          break;
+        case CountupQueryCalculation.maxPast:
+          values = values.filter((v: { time: Moment; description: string }) => v.time.isBefore(now));
+          _value = values.length > 0 ? sort_values(values).at(-1) : undefined;
+          break;
+        default:
+          console.error('Invalid query calculation', clockSettings.queryCalculation);
```

Comment:
I only did that because you do the same in [RenderZone.tsx](https://github.com/grafana/clock-panel/blame/e342be1aa37ea63d42aa6631448b863169550a2b/src/components/RenderZone.tsx#L37) - also as a fallback in a switch statement. My console.log should not happen, it's only triggering if someone adds QueryCalculations without adding cases to the switch.

Reply:
I only did that because you did the same in [RenderZone.tsx](https://github.com/grafana/clock-panel/blame/e342be1aa37ea63d42aa6631448b863169550a2b/src/components/RenderZone.tsx#L37) - also as a fallback in a switch statement. My console.log should not happen, it's only triggered if someone adds QueryCalculations without adding cases to the switch.